desc/objectives/_geometry.py

```diff
@@ -493,6 +493,346 @@ def compute(self, params, constants=None):
+        this should also be set to True.
+        be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: Has no effect for this objective.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    def __init__(
+        self,
+        coil,
+        data_keys,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name=None,
+    ):
+        self._grid = grid
+        self._data_keys = data_keys
+        super().__init__(
+            things=[coil],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        # local import to avoid circular import
+        # TODO: import this from desc backend?
+        import jax.tree_util as jtu
+
+        from desc.coils import CoilSet, MixedCoilSet, _Coil
+
+        coils = jtu.tree_flatten(
+            self.things[0],
+            is_leaf=lambda x: isinstance(x, _Coil) and not isinstance(x, CoilSet),
+        )[0]
+        self._dim_f = len(coils)
+
+        is_mixed_coils = isinstance(self.things[0], MixedCoilSet)
+
+        if self._grid is None:
+            # TODO: raise error if grid, transforms are not the same size as mixed coils
+            self._grid = (
+                LinearGrid(N=32)
+                if not is_mixed_coils
+                else [LinearGrid(N=32)] * self._dim_f
+            )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        if is_mixed_coils:
+            transforms = [
+                get_transforms(self._data_keys, obj=coils[i], grid=self._grid[i])
```

Comment:
@f0uriest using this with mixed?

Reply:
@kianorr I think something like this will work for getting transforms for nested coilsets. It assumes a `CoilSet` all use the same transform, but a `MixedCoilSet` will need different transforms for each member.

Right now this assumes the grid is the same for all of them, if we have different grids you can modify this to map over the trees of coils and grids simultaneously, see the `tree_map` documentation.

```python
transforms = jax.tree_util.tree_map(lambda x: get_transforms(self._data_keys, obj=x, grid=grid),
                                    self.coils,
                                    is_leaf=lambda x: isinstance(x, _Coil) and not isinstance(x, MixedCoilSet)
                                   )
```